packages/agent/src/tool/executor/toolPlugin.ts

```diff
@@ -26,6 +25,15 @@ export function createToolPlugin({
 
           reply.status(200).send(textOutput);
         } catch (error: any) {
+          const logger = pino({
+            level: "debug",
+            messageKey: "message",
+          });
```

Comment:
Can you make logger an additional argument of `createToolPlugin` and initialize it in the main server?